python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -314,6 +336,46 @@ def run(self):
             _, state = self._output_node
             state.mark_finished(exc)
 
+    def update_metrics(self, sched_loop_duration: int):
+        self._sched_loop_duration_s.set(
+            sched_loop_duration, tags={"dataset": self._dataset_id}
+        )
+        if self._initial_stats:
+            self._initial_stats.streaming_exec_schedule_s.add(sched_loop_duration)
```

Comment:
Nit: Would this make more sense out of `update_metrics()`? Felt kinda out of place to me since this is for stats and not Prometheus metrics